hathor/verification/transaction_verifier.py

```diff
@@ -121,7 +125,9 @@ def verify_inputs(self, tx: Transaction, *, skip_script: bool = False) -> None:
                 ))
 
             if not skip_script:
-                self.verify_script(tx=tx, input_tx=input_tx, spent_tx=spent_tx)
+                script_context = self.verify_script(tx=tx, spent_tx=spent_tx, input_index=input_index)
+                selected_outputs = script_context.get_selected_outputs()
+                all_selected_outputs = all_selected_outputs.union(selected_outputs)
```

Comment:
You can use `all_selected_outputs.update(selected_outputs)` to prevent creating a new set.